src/Shells/StrixMusic.Shells.ZuneDesktop/Controls/Views/Collection/ZuneTrackCollection.cs

```diff
@@ -16,154 +20,182 @@ namespace StrixMusic.Shells.ZuneDesktop.Controls.Views.Collection
     /// <summary>
     /// Zune implmenation for the <see cref="ZuneTrackCollection"/>.
     /// </summary>
-    public class ZuneTrackCollection : TrackCollection
+    [INotifyPropertyChanged]
+    public partial class ZuneTrackCollection : CollectionControl<ZuneTrackCollectionItem, ZuneTrackItem>
     {
+        private ObservableCollection<ZuneTrackCollectionItem> _trackItems = new();
+
+        public ZuneTrackCollection()
+        {
+            TrackItems = new ReadOnlyObservableCollection<ZuneTrackCollectionItem>(_trackItems);
+
+            Loaded += ZuneTrackCollection_Loaded;
+            Unloaded += ZuneTrackCollection_Unloaded;
```

Comment:
Subscribe the Unloaded Event after the control has loaded and then unload the loaded event there

Reply:
That's not needed. If the control is never loaded in XAML, we'll end up with a stray event that can't be unsubscribed either way, whether that's `Loaded` or `Unloaded`.

This is only used as a last chance sanity check for detaching events. I've added comments clarifying this.